tools/migration/main.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"runtime"
+
+	"github.com/filecoin-project/go-filecoin/tools/migration/internal"
+	"github.com/filecoin-project/go-filecoin/util/version"
+)
+
+func main() { // nolint: deadcode
+	if len(os.Args) < 2 {
+		showUsageAndExit(1)
+	}
+
+	if !version.Check(runtime.Version()) {
+		exitErr(fmt.Sprintf("Invalid go version: %s", runtime.Version()))
+	}
+
+	command := getCommand()
+
+	switch command {
+	case "-h", "--help":
+		showUsageAndExit(0)
+	case "describe", "buildonly", "migrate", "install":
+		mr := internal.MigrationRunner{}
+		if err := mr.Run(command, getVerbose()); err != nil {
+			exitErr(err.Error())
+		}
+	default:
+		exitErr(fmt.Sprintf("Error: Invalid command: %s", command))
+	}
+}
+
+// TODO: should this go to the same logs as the migration log? Maybe not.
+func exitErr(errstr string) {
+	log.New(os.Stderr, "", 0).Println(errstr)
+	os.Exit(1)
+}
+
+func showUsageAndExit(code int) {
+	fmt.Println(`    Usage:  go-filecoin-migrate (describe|buildonly|migrate) [--verbose][--oldRepo][--newRepo]`)
```

Comment:
Really like how simple this is.

Reply:
super small nit: it might be worth separating this out into a constant.  For example this would make it slightly easier to test as you wouldn't have to rewrite the constant in the tests.

Reply:
So, I debated separating it as a constant, however, I generally don't like pulling out constants as tests; I prefer forcing the test to manually match content, to help force extra proofreading, but also it makes it a little easier if there is string formatting (which is one of the things I was debating).